applications/dashboard/controllers/class.entrycontroller.php

```diff
@@ -468,9 +468,15 @@ public function Connect($Method) {
             $RoleIDs = $this->UserModel->NewUserRoleIDs();
          }
 
+         if (C('Garden.SSO.SyncRolesBehavior') === 'register') {
+            $SaveRolesRegister = TRUE;
+            $SaveRoles = FALSE;
+         }
+
```

Comment:
`Garden.SSO.SyncRolesBehavior` not being set is the current behavior. If `Garden.SSO.SyncRoles` is set and the behavior isn't then it means to sync roles all the time which it currently does.